pkg/module/parse/template_to_module.go

```diff
@@ -106,7 +108,69 @@ func (p *Parser) GenerateModulesFromTemplates(ctx context.Context,
 	return modules
 }
 
-func overwriteNameAndNamespace(template *channel.ModuleTemplateTO, name, namespace string) {
+func (p *Parser) GenerateMandatoryModulesFromTemplates(ctx context.Context,
```

Comment:
compare the logic of this method with `GenerateModulesFromTemplates`, lots of code is copy pasted, think about extract shared code into functions to increase the code reusability.